In [25]:
import pandas as pd

In [26]:
df = pd.read_csv('../data/raw/car_data.csv')

df.shape

(301, 9)

In [27]:
df.head(10)

,Car_Name,Year,Selling_Price,Present_Price,Driven_kms,Fuel_Type,Selling_type,Transmission,Owner
0,ritz,2014,3.35,5.59,27000,Petrol,Dealer,Manual,0
1,sx4,2013,4.75,9.54,43000,Diesel,Dealer,Manual,0
2,ciaz,2017,7.25,9.85,6900,Petrol,Dealer,Manual,0
3,wagon r,2011,2.85,4.15,5200,Petrol,Dealer,Manual,0
4,swift,2014,4.60,6.87,42450,Diesel,Dealer,Manual,0
5,vitara brezza,2018,9.25,9.83,2071,Diesel,Dealer,Manual,0
6,ciaz,2015,6.75,8.12,18796,Petrol,Dealer,Manual,0
7,s cross,2015,6.50,8.61,33429,Diesel,Dealer,Manual,0
8,ciaz,2016,8.75,8.89,20273,Diesel,Dealer,Manual,0
9,ciaz,2015,7.45,8.92,42367,Diesel,Dealer,Manual,0


In [28]:
df.columns

Index(['Car_Name', 'Year', 'Selling_Price', 'Present_Price', 'Driven_kms',
       'Fuel_Type', 'Selling_type', 'Transmission', 'Owner'],
      dtype='str')

In [29]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 301 entries, 0 to 300
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Car_Name       301 non-null    str    
 1   Year           301 non-null    int64  
 2   Selling_Price  301 non-null    float64
 3   Present_Price  301 non-null    float64
 4   Driven_kms     301 non-null    int64  
 5   Fuel_Type      301 non-null    str    
 6   Selling_type   301 non-null    str    
 7   Transmission   301 non-null    str    
 8   Owner          301 non-null    int64  
dtypes: float64(2), int64(3), str(4)
memory usage: 30.0 KB


In [30]:
df.isnull().sum()

Car_Name         0
Year             0
Selling_Price    0
Present_Price    0
Driven_kms       0
Fuel_Type        0
Selling_type     0
Transmission     0
Owner            0
dtype: int64

In [31]:
df['Car_Name'].value_counts()

Car_Name
city                   26
corolla altis          16
verna                  14
fortuner               11
brio                   10
                       ..
Hero Super Splendor     1
Hero Hunk               1
Hero  Ignitor Disc      1
Hero  CBZ Xtreme        1
Bajaj  ct 100           1
Name: count, Length: 98, dtype: int64

In [32]:
df['Year'].value_counts()

Year
2015    61
2016    50
2014    38
2017    35
2013    33
2012    23
2011    19
2010    15
2008     7
2009     6
2006     4
2005     4
2003     2
2007     2
2018     1
2004     1
Name: count, dtype: int64

In [33]:
df['Selling_Price'].describe()

count    301.000000
mean       4.661296
std        5.082812
min        0.100000
25%        0.900000
50%        3.600000
75%        6.000000
max       35.000000
Name: Selling_Price, dtype: float64

In [34]:
df['Present_Price'].describe()

count    301.000000
mean       7.628472
std        8.642584
min        0.320000
25%        1.200000
50%        6.400000
75%        9.900000
max       92.600000
Name: Present_Price, dtype: float64

In [35]:
df['Driven_kms'].describe()

count       301.000000
mean      36947.205980
std       38886.883882
min         500.000000
25%       15000.000000
50%       32000.000000
75%       48767.000000
max      500000.000000
Name: Driven_kms, dtype: float64

In [36]:
df['Fuel_Type'].value_counts(normalize=True)

Fuel_Type
Petrol    0.794020
Diesel    0.199336
CNG       0.006645
Name: proportion, dtype: float64

In [37]:
df['Selling_type'].value_counts(normalize=True)

Selling_type
Dealer        0.647841
Individual    0.352159
Name: proportion, dtype: float64

In [38]:
df['Transmission'].value_counts(normalize=True)

Transmission
Manual       0.86711
Automatic    0.13289
Name: proportion, dtype: float64

In [39]:
df['Owner'].value_counts()

Owner
0    290
1     10
3      1
Name: count, dtype: int64

In [40]:
print("Duplicate rows:", df.duplicated().sum())
print("Duplicate car+year combinations:", df.duplicated(subset=['Car_Name', 'Year']).sum())

Duplicate rows: 2
Duplicate car+year combinations: 95


**Duplicate findings:**

2 fully identical rows found — all 9 columns match exactly.
Decision: drop in cleaning notebook. Likely data entry errors.

95 rows share the same Car_Name + Year combination.
This is expected — multiple sellers listing the same model and year
with different prices, mileage, and conditions. These are legitimate
and will be kept.

In [41]:
current_year = df['Year'].max()  # use dataset's reference point
df_temp = 2018 - df['Year']  # age at time of most recent listing

print(f"Year range: {df['Year'].min()} to {df['Year'].max()}")
print(f"Cars manufactured over a {df['Year'].max() - df['Year'].min()} year span")
print(f"Age at listing — min: {(2018 - df['Year'].max())} yrs, max: {(2018 - df['Year'].min())} yrs")

Year range: 2003 to 2018
Cars manufactured over a 15 year span
Age at listing — min: 0 yrs, max: 15 yrs


## Data Understanding Summary

### Dataset
- 301 rows × 9 columns
- No missing values in any column
- Source: Indian used car market listings
- Prices are in Indian Lakhs (1 Lakh = 100,000 rupees)

### Columns
| Column | Type | Description |
|--------|------|-------------|
| Car_Name | string | Model name — 98 unique values, city is most common (26) |
| Year | int | Manufacturing year — 2003 to 2018 |
| Selling_Price | float | Price the owner is asking (Lakhs) — target variable |
| Present_Price | float | Current ex-showroom price of same model new (Lakhs) |
| Driven_kms | int | Total kilometres driven — wide range (500 to 500,000) |
| Fuel_Type | string | Petrol (79.4%), Diesel (19.9%), CNG (0.7%) |
| Selling_type | string | Dealer (64.8%) vs Individual (35.2%) |
| Transmission | string | Manual (86.7%) vs Automatic (13.3%) |
| Owner | int | Number of previous owners — 0=first, 1=second, 3=fourth |

### Target variable
Selling_Price — ranges from 0.1 to 35 Lakhs, mean 4.66, right-skewed.
Log transformation likely needed during modelling.

### Key observations
- Present_Price is always >= Selling_Price — used cars sell below new price, as expected
- Driven_kms has extreme outliers (max 500,000) — will need investigation in cleaning
- 96.3% of cars are first-owner — Owner column has very limited variance
- CNG fuel type has only 2 entries (0.7%) — may need to merge or drop in cleaning
- Dataset is small (301 rows) — model generalisation will be limited
- 2 fully identical duplicate rows found — will be dropped in cleaning
- 95 shared Car_Name + Year combinations — expected, legitimate separate listings
- Cars span 15 years of manufacture (2003–2018) — age at listing ranges from brand new (0 yrs) to 15 years old

### Next step
Exploratory data analysis — distributions, correlations, 
price depreciation by age and mileage, impact of fuel type and transmission.